sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -120,22 +140,36 @@ private WithRepresentativeValues(
       this.representativeType = representativeType;
     }
 
+
     @Override
     public PCollection<T> expand(PCollection<T> in) {
+      if (!(in.getWindowingStrategy().getWindowFn() instanceof NonMergingWindowFn)) {
+        LOG.warn("Distinct is not currently guaranteed to remove all duplicates when using merging"
+            + "window functions.");
+      }
+
       WithKeys<IdT, T> withKeys = WithKeys.of(fn);
       if (representativeType != null) {
         withKeys = withKeys.withKeyType(representativeType);
       }
-      return in
+      PCollection<KV<IdT, T>> combined = in
```

Comment:
The reason these are separate is because this stores a `T` as its accumulator while the other stores just a `VarInt`, right? It _might_ be worth generalizing them to share the same (private) implementation class, since they really have the same schema and consistent labels and validation would be nice.

At least pull out the validation.